flink-core/src/main/java/org/apache/flink/util/NetUtils.java

```diff
@@ -116,6 +119,39 @@ private static URL validateHostPortString(String hostPort) {
         }
     }
 
+    /**
+     * Calls {@link ServerSocket#accept()} on the provided server socket, suppressing any thrown
+     * {@link SocketTimeoutException}s. This is a workaround for the underlying JDK-8237858 bug in
+     * JDK 11 that can cause errant SocketTimeoutExceptions to be thrown at unexpected times.
+     *
+     * <p>This method expects the provided ServerSocket has no timeout set (SO_TIMEOUT of 0),
+     * indicating an infinite timeout. It will suppress all SocketTimeoutExceptions, even if a
+     * ServerSocket with a non-zero timeout is passed in.
+     *
+     * @param serverSocket a ServerSocket with {@link SocketOptions#SO_TIMEOUT SO_TIMEOUT} set to 0;
+     *     if SO_TIMEOUT is greater than 0, then this method will suppress SocketTimeoutException;
```

Comment:
We could. I started down this path, but I was not sure what the performance implications of calling `getSoTimeout` would be. The fact that it throws IOExceptions gave me pause. Given that there is only a single case where a socket is setting a timeout (in a test none the less) I felt it was safer to avoid the call altogether.

Granted, this landscape could change over time and this certainly would be a confusing bug. (suppressing timeouts unexpectedly) I named the method in a way that I hoped it would minimize the chances of this sort of thing.

I'm happy to go either way.

Reply:
>The fact that it throws IOExceptions gave me pause.

That's a fair point but it's probably not an issue. I'd wager a guess that this only fails if say, the socket is already closed. If getting an option throws an error the socket is probably unusable anyway 🤷 

Performance doesn't matter in this case; sockets aren't created on performance-critical paths.

So let's do something like
```
Preconditions.checkArgument(serverSocket.getSoTimeout() == 0, "<some reason>");
```

Reply:
Sounds good